File: iterate.py
import sys, time

def main():
    N = 1_000_000
    start = time.perf_counter()

    for i in range(1, N + 1):
        # Uncomment to spam numbers:
        print(i)
        if i % 100_000 == 0:
            elapsed = time.perf_counter() - start
            rate = i / elapsed if elapsed > 0 else 0.0
            eta = (N - i) / rate if rate > 0 else float("inf")
            print(f"\r{i}/{N}  elapsed={elapsed:.2f}s  ETA={eta:.2f}s", file=sys.stderr, end="", flush=True)

    # finish line
    print(file=sys.stderr)
    total = time.perf_counter() - start
    rate = N / total if total > 0 else 0.0
    print(f"Done. Elapsed={total:.3f}s  Avg={rate:,.0f} it/s", file=sys.stderr)

if __name__ == "__main__":
    main()

File: iterate.cpp
#include <iostream>
#include <chrono>
#include <iomanip>

int main() {
    using clock = std::chrono::high_resolution_clock;
    const int N = 1'000'000;
    auto t0 = clock::now();

    for (int i = 1; i <= N; ++i) {
        // Uncomment to spam numbers:
        // std::cout << i << '\n';
        if (i % 100000 == 0) {
            double elapsed = std::chrono::duration<double>(clock::now() - t0).count();
            double rate = (elapsed > 0) ? (i / elapsed) : 0.0;
            double eta  = (rate > 0) ? ((N - i) / rate) : 0.0;
            std::cerr << "\r" << i << "/" << N
                      << "  elapsed=" << std::fixed << std::setprecision(2) << elapsed << "s"
                      << "  ETA=" << std::fixed << std::setprecision(2) << eta << "s"
                      << std::flush;
        }
    }

    std::cerr << std::endl;
    double total = std::chrono::duration<double>(clock::now() - t0).count();
    double rate  = (total > 0) ? (N / total) : 0.0;
    std::cerr << "Done. Elapsed=" << std::fixed << std::setprecision(3) << total
              << "s  Avg=" << std::fixed << std::setprecision(0) << rate << " it/s\n";
    return 0;
}
